### ISSUE TITLE
Just getting started.

### ISSUE THREAD

**Issue by shahzeb1** [amilajack/electron-react-boilerplate-site#1]
Awesome, so just so I understand you would like the website to look something like the [Gravity](https://marcobambini.github.io/gravity/) website?

**Comment by amilajack** [comment]
Yup! I think its very clean and minimal, which is what I think we should go for. Ideally, we should have a consistent theme that matches the feel of the boilerplate's app as well. I think we should rewrite the boilerplate app to make it consistent with the website

**Comment by kamescg** [comment]
Can we have a sexy/graphic landing page and clean/minimal internal docs?

**Comment by amilajack** [comment]
Let's do that! Do you use sketch? I use it to design mockups.

**Comment by kamescg** [comment]
I'm happy to use either Sketch or Photoshop for UI. I'm excited! 

However, I am more proficient in Photoshop...

**Comment by amilajack** [comment]
I think we should base the design of the docs section off of [gravity lang's docs](https://marcobambini.github.io/gravity/index.html). The site will include:

* Video Tutorials
* Docs
* FAQ
* Guides

**Comment by kamescg** [comment]
I also really like Shopify's new docs (https://polaris.shopify.com/principles/principles)

**Comment by kamescg** [comment]
I think it would be great to feature Electron React Boilerplate Projects immediately on the site. In my opinion it really helps first-time visitors understand what's possible.

Multiple projects could be previewed in laptop screen?

![electron-react-boilerplate](https://cloud.githubusercontent.com/assets/3408362/25316299/7469f0e8-2819-11e7-8b28-39f813aa2fd9.jpg)

**Comment by amilajack** [comment]
> In my opinion it really helps first-time visitors understand what's possible.

Totally agree! [Here's the discussion on that](https://github.com/chentsulin/electron-react-boilerplate/issues/611)

**Comment by kamescg** [comment]
Nice - seems like a lot of the ideas are in place, we just need to start putting the pieces together?

**Comment by amilajack** [comment]
Yup! Also I think the landing page should have a small setup guide and features section, like [react-boilerplate's does](https://www.reactboilerplate.com)

**Comment by kamescg** [comment]
Are we using Github to collab on writing or something like Google Docs?

**Comment by amilajack** [comment]
Google Docs. [Here's the link](https://docs.google.com/document/d/1miZB8ch3cjD5jIUpaB3GyrkHe-P5_yjZU5-y0wfy_rw/edit?usp=sharing)

**Comment by kamescg** [comment]
Cool - I'll start adding content this week.

**Comment by kamescg** [comment]
I'm thinking we should delete the Git history, so we can start fresh for our project?

**Comment by amilajack** [comment]
Yup! Good idea.

**Comment by ansumanshah** [comment]
Awesome ideas here, and good design @KamesCG to start off with.
I would suggest that for the website we do not go forward with react-boilerplate rather go with something simple to setup and is ssr friendly like `next.js` and use `now` to push and test the website with ease. 

I like the approach the [styled-components](https://github.com/styled-components/styled-components-website) team are taking with their [site](https://sc-next-docs.philpl.com)/[docs](https://sc-next-docs.philpl.com/docs)
Simple setup and easy to update the site with minimal boilerplate.

Just an idea feel free to move forward with the current setup.
PS: I am a designer myself, where should I start with? Would be happy to set this up and get started as well just give me a heads up if we all agree.

one small addon it would be great if we add the website to the main repo itself will help get more contributors/issues from there.

**Comment by kamescg** [comment]
@ansumanshah I agree with a minimal and simple setup. I'm haven't looked at the code for the current repository, so I can be persuaded either direction. I like the idea of learning Next.js, and from what I saw, styled-components repository was very clean, and could almost immediately wrap my head around it.

I think we have a good understanding of the site-map, so we can  drafting mockups for each page?

Video Tutorials
Documentation
Frequently Asked Questions
Guides & Tutorials
Opinionated Boilerplate
Built With ERB

Missing anything? 
 
Might also be good to nail down the basics first, like color schemes, iconography, etc... because it seem like we have a pretty strong team here and can role-share. The blue in the Demo App might be a good place to start?

Another site I really like is egghead.io and especially the graphics.
![screen shot 2017-04-25 at 7 10 10 am](https://cloud.githubusercontent.com/assets/3408362/25389771/55c9b8b2-2986-11e7-983a-8a77cb483d16.png)

**Comment by amilajack** [comment]
@KamesCG I promise i'll finish the initial mockup for the design. I have a great theme in mind. Will post it here later today

**Comment by kamescg** [comment]
@amilajack Take it away 👍  Looking forward to seeing the design.

**Comment by kamescg** [comment]
In terms of the site structure it seems like we have a couple of of solid ideas/recommendations.

Personally, I was even thinking of recommending the Arc (https://arc.js.org) boilerplate, because of the close similarity with the Electron React Boilerplate structure: React, Redux, Webpack, Jest, etc... It might increase our ability to develop faster on both projects, because of the shared implementation. 

Code could live in the documentation and easily be ported to the boilerplate as needed. 

I'm of the opinion we should keep the site/boilerplate repositories separate. Even though it's been stated, we might get more contributors, I don't think that will necessarily be the case. Plus it adds unnecessary code and more files that needs to be removed for people to start with a clean build.

@amilajack Take the time to set the foundation for design and we can follow you lead.  I don't think we necessarily have to rush or compete to see who can pump out designs faster. In the meantime, over the next 2-3 weeks we'll focus on writing documentation and generating resources (slides, icons, bullet points, etc...) for screencasts. 

Last week I starting building a Dashboard Theme with Material UI. Since, we're all able to design/develop maybe it makes sense to tackle a couple of projects at the same time? First, so we can respect each contributions and not step on any toes. And second, we can solve several key challenges at the same time.... 🎉   asynchronously 🎉

I think it would be cool to incorporate a systems diagnostics and developer dashboard into a certain branch of the boilerplate. Maybe even project specific documentation automatically generated, available only in "dev" mode?
 
![screen shot 2017-04-25 at 10 27 58 pm](https://cloud.githubusercontent.com/assets/3408362/25419279/7e160dae-2a06-11e7-853d-d4ebf0e9c1d8.png)

**Comment by kamescg** [comment]
http://airbnb.design/painting-with-code/

The new React to Sketch tool looks promising.

**Comment by amilajack** [comment]
> Personally, I was even thinking of recommending the Arc

I havent used ARC before. What do you think of using [reactstrap](https://reactstrap.github.io/)? It provides react bindings to bootstrap components. We can use sass to customize the bootstrap components. Also I think using react bindings to bootstrap will make it easier for new contributors because they are probably already familiar with bootstrap.

> I don't think we necessarily have to rush or compete to see who can pump out designs faster.

Totally agree with you! Thanks!

> Maybe even project specific documentation automatically generated

I've heard Typescript as awesome tooling for this. Eventually I want to work on a codemod that can strip flow and migrate to Typescript.

**Comment by kamescg** [comment]
Reactstrap could be a great first UI library to get started with.

As I understand the situation we would probably want to use Reactstrap as the interface scaffolding components for Arc, which will give us essentials( redux, sagas, storybook, webpack, etc..)

However... what if we took it one step further and started with a **node/react content management system**?

https://github.com/relax/relax
![screen shot 2017-04-26 at 1 36 15 pm](https://cloud.githubusercontent.com/assets/3408362/25455623/5de8a54e-2a85-11e7-99b8-888358bb23bc.png)




You mentioned contributors, but I can't imagine we'll have to many designers/developers working on the general site layout infrastructure - that's why we have Reactstrap 👍  In my opinion, if anything, we should try and facilitate the space for contributors to more easily create an eco-system around Electron React Boilerplate.

A content management system would help lower the barrier to entry for people interested in contributing, but perhaps don't have time to clone another repository, fiddle with components and submit a pull request. At the same time, developers are still invited to work on the nuts'an-bolts.

- Community driven screencasts and tutorials about day-to-day challenges with Electron.
- A resource center for downloadable features and interface enhancements.
- Developer to developer conversations about best practices and challenges.

In terms of enforcing better Javascript practices using Typescript and Flow I'll leave that to the professionals :) I don't really have experience with these systems, so don't have a credible opinion. That being said, with Flow I like the idea of being able to gradually introduce a static type system as the application progresses. I can't fathom the thought of learning another paradigm at the moment...

**Comment by kamescg** [comment]
Want to schedule a skype/phone call to iron out the details?

**Comment by amilajack** [comment]
@KamesCG sure! @shahzeb1 what about you?

**Comment by amilajack** [comment]
@KamesCG Here's something i've come up with:

![v1](https://cloud.githubusercontent.com/assets/6374832/25561651/b14525bc-2d25-11e7-88ad-c25b9aea91da.jpg)

**Comment by amilajack** [comment]
Will upload the sketch file to this repo

**Comment by amilajack** [comment]
Added the design to `./design`. Sorry for the delay on this by the way.

**Comment by kamescg** [comment]
I think it's great! Simple, to the point and easy to understand. Awesome job 👍 

Life happens - we all have things to do :)

**Comment by kamescg** [comment]
In terms of site capabilities overtime, what types of features should we account for when thinking about the site infrastructure?

I really do think it's important to facilitate a community of users who can easily share best-practices, design patterns and easily shares "features" from project to project.

**Comment by shahzeb1** [comment]
Unfortunately I don't have sketch. 😭 

 Could someone please post a screenshot or two of the design?

**Comment by amilajack** [comment]
See [this comment](https://github.com/amilajack/electron-react-boilerplate-site/issues/1#issuecomment-298211180)

**Comment by shahzeb1** [comment]
On it.

Do you still think this electron website boilerplate is appropriate for such a small non-reactive website?

**Comment by amilajack** [comment]
Good question. Let's start with vanilla html and CSS for now. But let's keep the boilerplate for now. I feel like the boilerplate its useful for things like docs, which will probably involve a lot of routing logic and state changes. I dont think we should use redux because i dont think its necessary for this project.

**Comment by kamescg** [comment]
@amilajack I agree Redux would be overkill. I think Node/React in general is always overkill for this type of stuff anyways.

If we want to go static html/css for right now what about Grav (https://getgrav.org/) - it's just a flat-file CMS, with Markdown as the WYSIWYG. Would be super easy to setup, edit and add documentation.

**Comment by kamescg** [comment]
Here is a "skeleton" we can use http://demo.hibbittsdesign.org/grav-learn2-git-sync/

**Comment by amilajack** [comment]
@KamesCG great! I'll try to push an initial prototype!

**Comment by kamescg** [comment]
It requires PHP, so it can't be published in a Git project for free. Something like DigitalOcean maybe?

**Comment by amilajack** [comment]
I'm thinking of something similar to [gravity-lang's docs: just pure html, css](https://github.com/marcobambini/gravity/tree/master/docs). I don't think we'll have enough content to have a CMS. If we face scalability issues, lets migrate to a CMS

**Comment by kamescg** [comment]
I understand the perspective of keeping the project clean and simple.

From the perspective of someone who is interested in writing guides/tutorials for the boilerplate I'm not so thrilled at the idea of managing my content writing in plan HTML/CSS though. It would be nice to have some basic tools that make the writing process a little easier and more organized.

**Comment by amilajack** [comment]
What do you think about something like gitbook? I've found it to be rather easy to use. It also doesn't require that we need to know PHP/some other language.

**Comment by kamescg** [comment]
![0mkxcg1](https://cloud.githubusercontent.com/assets/3408362/25628715/1cbe7698-2f1c-11e7-8f48-7308e69a3db6.gif)

**Comment by amilajack** [comment]
Agreed! We can simply commit `.md` files and have the site trigger the updates to the website. I think it also has nice GitHub pages integration. What do you think?

**Comment by kamescg** [comment]
Sounds like a very reasonable solution to me.

**Comment by shahzeb1** [comment]
So what is the final consensus? What tool / platform / method are we sticking to?

**Comment by kamescg** [comment]
It would seem Gitbook is the final contender.

**Comment by amilajack** [comment]
Yup!

**Comment by shahzeb1** [comment]
What came of this?

**Comment by amilajack** [comment]
Still have yet to implement this

**Comment by kamescg** [comment]
Let's get it poppin!

I just started a new Electron Boilerplate project yesterday - https://github.com/kamescg/mineshaft
